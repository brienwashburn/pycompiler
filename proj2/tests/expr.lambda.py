lambda: 1
lambda x: x
lambda x,: x
lambda x=1: x
lambda x,y,z: x+y/z
lambda x,y,z=10: x+y/z

lambda x,*z,a=10: x
lambda x=1,*z,a: a
lambda x=1,*z,a=3: a
lambda x=1,*z,a,b: a
lambda x=1,*z,a=3,b: a
lambda x=1,*z,a,b=3: a
lambda x=1,*z,a=10,b=3: a
lambda x,*z,a,b=3: a
lambda x,y,z=1,*foo,a=18: a
lambda x,y=(2+3),z=1,*foo,a: a
lambda x,y=(2+3),z=10,*foo,a,**rest: rest
lambda x,y=(2+3),z=10,*foo,a=20,**rest: rest

lambda x,*,a=10: x
lambda x=1,*,a: a
lambda x=1,*,a=3: a
lambda x=1,*,a,b: a
lambda x=1,*,a=3,b: a
lambda x=1,*,a,b=3: a
lambda x=1,*,a=10,b=3: a
lambda x,*,a,b=3: a
lambda x,y,z=1,*,a=18: a
lambda x,y=(2+3),z=1,*,a: a
lambda x,y=(2+3),z=10,*,a,**rest: rest
lambda x,y=(2+3),z=10,*,a=20,**rest: rest

lambda *z,a: a
lambda *z,a=10: a
lambda *z,a,b: a
lambda *z,a=10,b: a
lambda *z,a,b=20: a
lambda *z,a=10,b=10: a
lambda *z,a,b=4,**rest: 2
lambda *z,a,b,**rest: 2
lambda *z,a=10,**rest: 2

lambda *,a: a
lambda *,a=10: a
lambda *,a,b: a
lambda *,a=10,b: a
lambda *,a,b=20: a
lambda *,a=10,b=10: a
lambda *,a,b=4,**rest: 2
lambda *,a,b,**rest: 2
lambda *,a=10,**rest: 2

lambda **rest: 3

# This one revelead a bug in pyparse once upon a time
lambda x=3, y=4, *z: 0
