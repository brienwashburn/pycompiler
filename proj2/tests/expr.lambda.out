(Module
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults))
   (Num 1)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f))
   (Name x)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f))
   (Name x)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults (Num 1)))
   (Name x)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f #f #f))
   (BinOp (Name x) Add (BinOp (Name y) Div (Name z)))))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults #f #f (Num 10)))
   (BinOp (Name x) Add (BinOp (Name y) Div (Name z)))))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 10))
    (kwarg #f)
    (defaults #f))
   (Name x)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 3))
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f #f)
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 3) #f)
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 3))
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 10) (Num 3))
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 3))
    (kwarg #f)
    (defaults #f))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg foo)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 18))
    (kwarg #f)
    (defaults #f #f (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg foo)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults #f (BinOp (Num 2) Add (Num 3)) (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg foo)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg rest)
    (defaults #f (BinOp (Num 2) Add (Num 3)) (Num 10)))
   (Name rest)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg foo)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 20))
    (kwarg rest)
    (defaults #f (BinOp (Num 2) Add (Num 3)) (Num 10)))
   (Name rest)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 10))
    (kwarg #f)
    (defaults #f))
   (Name x)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 3))
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f #f)
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 3) #f)
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 3))
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 10) (Num 3))
    (kwarg #f)
    (defaults (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x)
    (arg-types #f)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 3))
    (kwarg #f)
    (defaults #f))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 18))
    (kwarg #f)
    (defaults #f #f (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults #f (BinOp (Num 2) Add (Num 3)) (Num 1)))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg rest)
    (defaults #f (BinOp (Num 2) Add (Num 3)) (Num 10)))
   (Name rest)))
 (Expr
  (Lambda
   (Arguments
    (args x y z)
    (arg-types #f #f #f)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 20))
    (kwarg rest)
    (defaults #f (BinOp (Num 2) Add (Num 3)) (Num 10)))
   (Name rest)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 10))
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f #f)
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 10) #f)
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 20))
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 10) (Num 10))
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 4))
    (kwarg rest)
    (defaults))
   (Num 2)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f #f)
    (kwarg rest)
    (defaults))
   (Num 2)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg z)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 10))
    (kwarg rest)
    (defaults))
   (Num 2)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults #f)
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 10))
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f #f)
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 10) #f)
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 20))
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults (Num 10) (Num 10))
    (kwarg #f)
    (defaults))
   (Name a)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f (Num 4))
    (kwarg rest)
    (defaults))
   (Num 2)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a b)
    (kwonlyarg-types #f #f)
    (kw_defaults #f #f)
    (kwarg rest)
    (defaults))
   (Num 2)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs a)
    (kwonlyarg-types #f)
    (kw_defaults (Num 10))
    (kwarg rest)
    (defaults))
   (Num 2)))
 (Expr
  (Lambda
   (Arguments
    (args)
    (arg-types)
    (vararg #f)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg rest)
    (defaults))
   (Num 3)))
 (Expr
  (Lambda
   (Arguments
    (args x y)
    (arg-types #f #f)
    (vararg z)
    (kwonlyargs)
    (kwonlyarg-types)
    (kw_defaults)
    (kwarg #f)
    (defaults (Num 3) (Num 4)))
   (Num 0))))
